pkg/pbflow/proto.go

```diff
@@ -0,0 +1,202 @@
+package pbflow
+
+import (
+	"encoding/binary"
+	"net"
+
+	"github.com/netobserv/netobserv-ebpf-agent/pkg/ebpf"
+	"github.com/netobserv/netobserv-ebpf-agent/pkg/flow"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// FlowsToPB is an auxiliary function to convert flow records, as returned by the eBPF agent,
+// into protobuf-encoded messages ready to be sent to the collector via GRPC
+func FlowsToPB(inputRecords []*flow.Record, maxLen int) []*Records {
+	entries := make([]*Record, 0, len(inputRecords))
+	for _, record := range inputRecords {
+		entries = append(entries, FlowToPB(record))
+	}
+	var records []*Records
+	for len(entries) > 0 {
+		end := len(entries)
+		if end > maxLen {
+			end = maxLen
+		}
+		records = append(records, &Records{Entries: entries[:end]})
+		entries = entries[end:]
+	}
+	return records
+}
+
+// FlowToPB is an auxiliary function to convert a single flow record, as returned by the eBPF agent,
+// into a protobuf-encoded message ready to be sent to the collector via kafka
+func FlowToPB(fr *flow.Record) *Record {
```

Comment:
curious why did u move this file from export to here ?

Reply:
it's also called from `decode` package now, and leaving it in `export` would create a cycle. The other option would be to move it to `decode`, I can do that if you prefer.